modules/common/labels/labels.go

```diff
@@ -86,3 +86,25 @@ func GetAppLabelSelector(
 ) metav1.LabelSelector {
 	return GetSingleLabelSelector(common.AppSelector, name)
 }
+
+// GetLabelSelector - utility function that returns a metav1.LabelSelector
+// based on the map[string]string that represents the k/v list passed to the
+// StatefulSet or Deployment as labelSelector
+func GetLabelSelector(
+	serviceLabels map[string]string,
+) metav1.LabelSelector {
+	lsr := []metav1.LabelSelectorRequirement{}
+	for key, value := range serviceLabels {
+		item := metav1.LabelSelectorRequirement{
+			Key:      key,
+			Operator: metav1.LabelSelectorOpIn,
+			Values: []string{
+				value,
+			},
+		}
+		lsr = append(lsr, item)
+	}
+	return metav1.LabelSelector{
+		MatchExpressions: lsr,
```

Comment:
I don't think this is necessary, can just do: 
`
metav1.LabelSelector {
    MatchLabels: serviceLabels
}
`

Reply:
Ah thank you, that is definitely simpler and should have the same behavior. Updating the patch! Thank you!

Reply:
Done

Reply:
I can see the statefulSet being updated with:

```
...
      topologySpreadConstraints:
      - labelSelector:
          matchLabels:
            component: glance-api
            glanceAPI: glance-default-internal
            owner: glance-default-internal
            service: glance
```